xgcm/grid.py

```diff
@@ -569,14 +573,21 @@ def cumsum(self, da, axis, **kwargs):
         return ax.cumsum(da, **kwargs)
 
 
-def add_slice(da, dim, sl, added):
-    axis = da.get_axis_num(dim)
-    t_order = list(range(len(da.data.shape)))
-    t_order[axis] = 0
-    t_order[0] = axis
-    data = np.transpose(da.data.copy(), axes=t_order)
-    data[sl, ...] = data[sl, ...] + added
-    return np.transpose(data, t_order)
+def add_to_slice(da, dim, sl, value):
+    # split array into before, middle and after (if slice is the
+    # beginning or end before or after will be empty)
+    before = da[{dim: slice(0, sl)}]
+    middle = da[{dim: sl}]
+    after = da[{dim: slice(sl+1, None)}]
```

Comment:
perfect...this should be dask friendly

Reply:
is there a preferred way to test if a computation was triggered?